### ISSUE TITLE
Is it possible to train using only an image dataset?

### ISSUE THREAD

**Issue by fhsqqa7** [AkaneTendo25/musubi-tuner#78]
When training LTX 2.3 character LoRA using only images, it appears to have almost no effect.

When training for 3,000 steps with the settings below, looks like a completely different person.
Am I using incorrect configs?
(When checking the logs, the loss fluctuates but does not decrease.)

However, when training for 3,000 steps on WAN with the same dataset, it shows high similarity.

====

### dataset: 
33 images

### dataset.toml:
[general]
resolution = [768,768]
caption_extension = ".txt"
batch_size = 1
enable_bucket = true
bucket_no_upscale = false
[[datasets]]
image_directory = "/workspace/dataset"
num_repeats = 1

### caching:
python ltx2_cache_latents.py \
  --dataset_config ... \
  --ltx2_checkpoint ... \
  --device cuda \
  --vae_dtype bf16 \
  --ltx2_mode v \
  --ltx2_audio_source video

python ltx2_cache_text_encoder_outputs.py \
  --dataset_config ... \
  --ltx2_checkpoint ... \
  --gemma_safetensors ... \
  --device cuda \
  --mixed_precision bf16 \
  --ltx2_mode v \
  --batch_size 1

### training conf:
--num_cpu_threads_per_process 1 ltx2_train_network.py
--mixed_precision bf16
--dataset_config /workspace/dataset/...
--ltx2_checkpoint /workspace/models/...
--ltx_version 2.3
--ltx_version_check_mode error
--ltx2_mode v
--gemma_safetensors /workspace/models/...
--separate_audio_buckets
--sdpa
--fp8_base
--fp8_scaled
--blocks_to_swap 10
--optimizer_type adamw8bit
--optimizer_args weight_decay=0.05
--gradient_checkpointing
--network_module networks.lora_ltx2
--timestep_sampling shifted_logit_normal
--max_train_steps 3000
--save_every_n_steps 250
--seed 42
--gradient_accumulation_steps=1
--output_dir /workspace/output
--output_name model
--logging_dir /workspace/log
--log_with tensorboard
--learning_rate 1e-4
--lr_scheduler constant
--network_dim 32
--network_alpha 32
--network_dropout 0.05

**Comment by AkaneTendo25** [comment]
Yes, it's possible, at least it definitely worked with LTX 2.0. I did not test image-only training with LTX 2.3, so I don't know what the optimal parameters are for training on images only.

You can try setting `--lora_target_preset video_sa_ca_ff` to strengthen the effect of LoRA. By default (`t2v` preset), it trains attention-only layers, so forcing feedforward layers into training may improve it.

Also maybe weight_decay=0.05 and network_dropout=0.05 are a bit too strong, you may try training without them.

**Comment by fhsqqa7** [comment]
> Yes, it's possible, at least it definitely worked with LTX 2.0. I did not test image-only training with LTX 2.3, so I don't know what the optimal parameters are for training on images only.
> 
> You can try setting `--lora_target_preset video_sa_ca_ff` to strengthen the effect of LoRA. By default (`t2v` preset), it trains attention-only layers, so forcing feedforward layers into training may improve it.
> 
> Also maybe weight_decay=0.05 and network_dropout=0.05 are a bit too strong, you may try training without them.

Thank you for your advice. 
When retraining with the settings you suggested, I was able to confirm that the similarity increased compared to before, and the loss also decreased as the steps progressed.

However, perhaps because LTX itself requires many steps for training, the similarity does not reach the level of WAN even after running up to 5,000 steps.

I am not sure if a larger dataset or different training settings are needed, but I do not have the luxury of time.🥲

**Comment by pravbk100** [comment]
> > Yes, it's possible, at least it definitely worked with LTX 2.0. I did not test image-only training with LTX 2.3, so I don't know what the optimal parameters are for training on images only.
> > You can try setting `--lora_target_preset video_sa_ca_ff` to strengthen the effect of LoRA. By default (`t2v` preset), it trains attention-only layers, so forcing feedforward layers into training may improve it.
> > Also maybe weight_decay=0.05 and network_dropout=0.05 are a bit too strong, you may try training without them.
> 
> Thank you for your advice. When retraining with the settings you suggested, I was able to confirm that the similarity increased compared to before, and the loss also decreased as the steps progressed.
> 
> However, perhaps because LTX itself requires many steps for training, the similarity does not reach the level of WAN even after running up to 5,000 steps.
> 
> I am not sure if a larger dataset or different training settings are needed, but I do not have the luxury of time.🥲

Depends on your dataset. I too had same problems like lora not converging fast. so I increased batch size in dataset.toml. My advantage was i had 1000 face only images(different angle, different lighting etc) at 256 resolution. so i made batch size of 10. And i added only feed forword in the network args rather than normal qkv. and the lora started converging at around 1000 steps, overbake at around 1500 steps. Just one word captions or trigger word as captions for all. 

If you want more advanced and even faster training, then the character identity mainly lives in between 20-40 blocks. you can target those only too.

**Comment by girzai** [comment]
> Yes, it's possible, at least it definitely worked with LTX 2.0. I did not test image-only training with LTX 2.3, so I don't know what the optimal parameters are for training on images only.
> 
> You can try setting `--lora_target_preset video_sa_ca_ff` to strengthen the effect of LoRA. By default (`t2v` preset), it trains attention-only layers, so forcing feedforward layers into training may improve it.
> 
> Also maybe weight_decay=0.05 and network_dropout=0.05 are a bit too strong, you may try training without them.

Hi there, I trained a lora with images and Im satisfied with the result, btw the lora make mute video ( as I train with images ) , it's they're a parameter for dont train the audio part ? Thanks

**Comment by AkaneTendo25** [comment]
If you use `--ltx2_mode video` and `--lora_target_preset video_sa_ca_ff`, audio modules aren't targeted, so the LoRA shouldn't affect it. The audio should just stay the same as the base model. Or are you already using those and the audio is still broken?
